omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_4/OrderSearchHandler2_4.java

```diff
@@ -0,0 +1,220 @@
+	            REQUEST_PARAM_EXCLUDE_CANCELED_AND_EXPIRED,
+	            REQUEST_PARAM_INCLUDE_VOIDED).build();
+	
+	private final SearchConfig searchConfig = new SearchConfig("default", RestConstants.VERSION_1
+	        + "/order", Arrays.asList("2.4.*"), searchQuery);
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.SearchHandler#getSearchConfig()
+	 */
+	@Override
+	public SearchConfig getSearchConfig() {
+		return searchConfig;
+	}
+	
+	/**
+	 * @see SearchHandler#search(RequestContext)
+	 */
+	@Override
+	public PageableResult search(RequestContext context) throws ResponseException {
+		// get input parameters
+		String patientUuid = context.getParameter("patient");
+		String careSettingUuid = context.getParameter("careSetting");
+		String conceptUuids = context.getParameter("concepts");
+		String orderTypeUuids = context.getParameter("orderTypes");
+		String activatedOnOrBeforeDateStr = context.getParameter("activatedOnOrBeforeDate");
+		String activatedOnOrAfterDateStr = context.getParameter("activatedOnOrAfterDate");
+		String isStoppedStr = context.getParameter(REQUEST_PARAM_IS_STOPPED);
+		String autoExpireOnOrBeforeDateStr = context.getParameter(REQUEST_PARAM_AUTO_EXPIRE_ON_OR_BEFORE_DATE);
+		String canceledOrExpiredOnOrBeforeDateStr = context
+		        .getParameter(REQUEST_PARAM_CANCELED_OR_AUTO_EXPIRE_ON_OR_BEFORE_DATE);
+		String actionStr = context.getParameter(REQUEST_PARAM_ACTION);
+		String fulfillerStatusStr = context.getParameter(REQUEST_PARAM_FULFILLER_STATUS);
+		String excludeCanceledAndExpiredStr = context.getParameter(REQUEST_PARAM_EXCLUDE_CANCELED_AND_EXPIRED);
+		String includeVoidedStr = context.getParameter("includeVoided");
+		
+		// build search criteria for order service
+		boolean includeVoided = StringUtils.isNotBlank(includeVoidedStr) ? Boolean.parseBoolean(includeVoidedStr) : false;
+		// by default the Canceled(dateStopped != null) and Expired(autoExpire < today) orders are excluded
+		boolean excludeCanceledAndExpired = StringUtils.isNotBlank(excludeCanceledAndExpiredStr) ? Boolean
+		        .parseBoolean(excludeCanceledAndExpiredStr) : false;
+		boolean isStopped = StringUtils.isNotBlank(isStoppedStr) ? Boolean.parseBoolean(isStoppedStr) : false;
+		
+		Date activatedOnOrBeforeDate = StringUtils.isNotBlank(activatedOnOrBeforeDateStr) ?
+		        (Date) ConversionUtil.convert(activatedOnOrBeforeDateStr, Date.class) : null;
+		Date activatedOnOrAfterDate = StringUtils.isNotBlank(activatedOnOrAfterDateStr) ?
+		        (Date) ConversionUtil.convert(activatedOnOrAfterDateStr, Date.class) : null;
+		Date autoExpireOnOrBeforeDate = StringUtils.isNotBlank(autoExpireOnOrBeforeDateStr) ?
+		        (Date) ConversionUtil.convert(autoExpireOnOrBeforeDateStr, Date.class) : null;
+		Date canceledOrExpiredOnOrBeforeDate = StringUtils.isNotBlank(canceledOrExpiredOnOrBeforeDateStr) ?
+		        (Date) ConversionUtil.convert(canceledOrExpiredOnOrBeforeDateStr, Date.class) : null;
```

Comment:
I think you can move the date validation logic into a single method and call that when you need to read the date.